bindings/dart/lib/src/web/worker/message/response.dart

```diff
@@ -14,13 +16,18 @@ part 'response.g.dart';
 @JsonSerializable()
 class Response implements ToJson {
   final Map<String, dynamic>? result;
+  final Map<String, dynamic>? exception;
```

Comment:
it would be nice if dart had stateful enums :/ we have to be very careful to not set both fields not null. i think the code is small enough that the current solution works. otherwise we could also rewrite it a bit similarly to a rust `Result<T, E>`, what do you think of the following?

```dart
enum Result {
  ok,
  exception,
}
```

```suggestion
  final Map<String, dynamic>? result;
  final Result kind;
```

Reply:
Good idea! I replaced exception with the Result enum